android/src/main/cpp/JNIHelper.cpp

```diff
@@ -52,8 +52,47 @@ jni::local_ref<JNIHelper::PropsMap> JNIHelper::ConvertToPropsMap(
       }
     }
   }
-
   return map;
 }
 
+jsi::Object JNIHelper::convertJNIMapToJSIObject(
```

Comment:
I don't quite understand why the conversion is part of this PR. Why do we need it with progress animation and didn't need this before? Seem to me like the progress animation shouldn't add a lot of additional complexity as this is just a single progress number that we are dealing with